### DISCUSSION TITLE
How can I launch more than one bot from a project?

### DISCUSSION THREAD

**Discussion by iamsisa** [aiogram/aiogram#758] [Q&A]
bot = Bot(token='token1',parse_mode = 'HTML')
dp = Dispatcher(bot,storage = MemoryStorage())

bot1 = Bot(token='token2',parse_mode = 'HTML')
dp1 = Dispatcher(bot1,storage = MemoryStorage())

I have two bots, tell me how to run them in parallel

**Comment by iamsisa** [comment]
@Olegt0rr help

**Reply by Olegt0rr** [reply]
You should run polling as a task for each one

**Reply by iamsisa** [reply]
I do so, but only the first token is triggered, the rest do not work.

**Reply by Olegt0rr** [reply]
It seems you did something wrong :)

**Reply by iamsisa** [reply]
https://pastebin.com/Tc3kXuVr
What am I doing wrong? Only dp responds, and dp1 does not want to

**Reply by evgfilim1** [reply]
@iamsisa Please don't:
- Screenshot code. Paste it here by wrapping into three ``` or use some paste services like gist or pastebin
- Post your bot tokens anywhere

**Reply by iamsisa** [reply]
@evgfilim1 done!

**Comment by Olegt0rr** [comment]
> https://pastebin.com/Tc3kXuVr What am I doing wrong? Only dp responds, and dp1 does not want to

Don't use threads for this.
Use asyncio tasks.
Use polling from dispatcher, not from executor.

**Comment by humza-sami** [comment]
I know its quite late but i fixed it like this:

 ```python
 import asyncio

  async def start_raid_bot():
      dp_bot1 = Dispatcher()
      dp_bot1.include_router(raid_router)
      await dp_bot1.start_polling(bot1)
  
  async def start_trend_bot():
      dp_bot2 = Dispatcher()
      dp_bot2.include_router(router_trend_bot)
      await dp_bot2.start_polling(bot2)
  
  async def main():
      asyncio.create_task(confirm_payment_and_inform_user())
      task_bot1 = asyncio.create_task(start_raid_bot())
      task_bot2 = asyncio.create_task(start_trend_bot())
      await task_bot1
      await task_bot2
  
  if __name__ == "__main__":
      asyncio.run(main())

```
